electronics_lib/PowerConditioning.py

```diff
@@ -263,3 +264,136 @@ def connected_from(self, gnd: Optional[Port[VoltageLink]] = None, pwr_hi: Option
     if pwr_lo is not None:
       cast(Block, builder.get_enclosing_block()).connect(pwr_lo, self.pwr_lo)
     return self
+
+
+class PmosReverseProtection(PowerConditioner, KiCadSchematicBlock, Block):
+  """
+  - This circuit uses a PMOS as a switch, which turns on when the power supply is correctly connected,
+  allowing current to flow to the load. If the power supply polarity is reversed, the PMOS turns off,
+  disconnecting the load and protecting the circuit.
+  - This method is preferred over diode-based protection due to lower power loss
+  - In most cases, 100R-330R is good if there are chances for the appearance of sudden reverse voltage in the circuit.
+  - But if there are no chances of sudden reverse voltage during the continuous working of the circuit, anything from the 1k-50k resistor value can be used.
+  - Ref: https://components101.com/articles/design-guide-pmos-mosfet-for-reverse-voltage-polarity-protection
+  """
+  @init_in_parent
+  def __init__(self, gate_resistor: RangeLike = 10 * kOhm(tol=0.05),):
```

Comment:
Should this also allow a user-specified Rds,on like the charging-capable variant?

Reply:
![image](https://github.com/BerkeleyHCI/PolymorphicBlocks/assets/49264928/360a950e-da56-4489-929c-cf8a5598384e)

Adding rds_on raised this error, but is rds_on set?  I tried with specific params like (0, 0.1)ohm, but still the same error.

```python
  @init_in_parent
  def __init__(self, gate_resistor: RangeLike = 10 * kOhm(tol=0.05), rds_on: RangeLike = Range.all()):
    super().__init__()
    self.gnd = self.Port(Ground.empty(), [Common])
    self.pwr_in = self.Port(VoltageSink.empty())
    self.pwr_out = self.Port(VoltageSource.empty())

    self.gate_resistor = self.ArgParameter(gate_resistor)
    self.rds_on = self.ArgParameter(rds_on)

  def contents(self):
    super().contents()
    output_current_draw = self.pwr_out.link().current_drawn
    self.fet = self.Block(Fet.PFet(
      drain_voltage=(0, self.pwr_out.link().voltage.upper()),
      drain_current=output_current_draw,
      gate_voltage=(- self.pwr_out.link().voltage.upper(), self.pwr_out.link().voltage.upper()),
      rds_on=self.rds_on,
    ))
    
    ```

Reply:
with (0, 0.1)Ohm try clean rebuild the block

Reply:
Resolved with [979177d](https://github.com/BerkeleyHCI/PolymorphicBlocks/pull/328/commits/979177d271ac08b925111a06172a237db55e7c3c)